Sources_v3/Database/DTOs/DeviceDTO_Tests.swift

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient
+import XCTest
+
+class DeviceDTO_Tests: XCTestCase {
+    var database: DatabaseContainer!
+    
+    override func setUp() {
+        super.setUp()
+        database = try! DatabaseContainer(kind: .inMemory)
+    }
+    
+    func test_devicesPayload_isStoredAndLoadedFromDB() throws {
+        let dummyDevices = dummyDevicesPayload
+        
+        try database.writeSynchronously { (session) in
+            // Save a current user to db for testing
+            try session.saveCurrentUser(payload: self.dummyCurrentUser)
+            
+            // Save dummy devices
+            try session.saveCurrentUserDevices(payload: dummyDevices)
+        }
+        
+        // Get current user from DB
+        let loadedCurrentUser: CurrentChatUser? = database.viewContext.currentUser()?.asModel()
+        
+        // Check if fields are correct
+        XCTAssertEqual(loadedCurrentUser?.devices.count, 2)
+        let sortedCurrentUserDevices = loadedCurrentUser?.devices.sorted(by: { $0.id > $1.id })
+        let sortedDummyDevices = dummyDevices.devices.sorted(by: { $0.id > $1.id })
+        XCTAssertEqual(sortedCurrentUserDevices?.first?.id, sortedDummyDevices.first?.id)
+        XCTAssertEqual(sortedCurrentUserDevices?.first?.createdAt, sortedDummyDevices.first?.createdAt)
+    }
+}
+
+extension XCTestCase {
+    var dummyDevice: DevicePayload {
+        .init(id: .unique, createdAt: .unique)
+    }
+    
+    var dummyDevicesPayload: DevicesPayload {
+        .init(devices: [dummyDevice, dummyDevice])
+    }
+}
```

Comment:
Why not define these methods as an extension of `DevicePayload`, and ideally in `Tests_v3/Dummy Data` like we do with the others?

Reply:
I missed those, I was referring dummy payloads defined in `ChannelDTO_Tests`

Reply:
Fixed in 46f926451006e1b0be9125afc4109583cef1a9f7